sound/soc/sof/intel/hda-pcm.c

```diff
@@ -140,6 +147,36 @@ int hda_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+/* update SPIB register with appl position */
+int hda_dsp_pcm_ack(struct snd_sof_dev *sdev, struct snd_pcm_substream *substream)
+{
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(hstream);
+	struct snd_pcm_runtime *runtime = substream->runtime;
+	ssize_t appl_pos, buf_size;
+	u32 spib;
+
+	/*
+	 * if a rewind request took place after the RUN bits were programmed,
+	 * deny it since hardware only supports monotonic (modulo) increments for SPIB.
+	 */
+	if (hstream->running && hda_disable_rewinds && substream->rewind_requested)
```

Comment:
This looks a bit clunky, but seems like a necessary evil here. I definitely agree having a simple check and return EINVAL is better. If we have some complex code here that tries to keep rewinds working, it could cause hard-to-track audio issues and be a pain to debug.

Reply:
Yes we don't even want to go there ... Better fail big and early than try to figure out what happened.